### DISCUSSION TITLE
What is the best stack for high performance of terminal server?

### DISCUSSION THREAD

**Discussion by meklonets** [neutrinolabs/xrdp#2814] [Q&A]
Hello.
Trying to migrate from Windows RDS with 30 users to a Linux server.
The Windows server is a virtual machine allocated at the QEMU/KVM/libvirtd hypervisor on a Arch Linux host. The Linux VM as well. Both work with a virtio and the host's cpu-passthrough enabled.
Is there any recommendation what stack is the best for performance? I mean on what OS (ubuntu, ubuntu LTS, ubuntu server or debian), what DE (xfce, cinnamon, lxde) to use? And maybe some instruction to tune xrdp, if any?
I tested lxde and cinnamon on Debian 12. Started 10 user's sessions and launch blank page of chromium on every session. All 4 vCPU (i9-11900K) of Linux VM went 100% load. So even ssh connection to the VM got frozen. Actually the freeze started when I launched chromium on the 6th session in a raw. Disconnecting (not logging off) all the sessions helped - vCPU set free.
Tried Libreoffice instead the chromium and the situation is the same.

**Comment by matt335672** [comment]
First point; high CPU load could also be indicative of a memory shortage in these days of SSDs. Check you're not starting to hit the swap space. A frozen ssh session sounds like more than a simple CPU overload.

Use a simple desktop like XFCE without a compositing window manager. Colour effects are probably not necessary and can cause a lot of traffic.

Make sure the (Windows?) clients are set to 32-bit colour and the connection type is set to LAN rather than auto. Windows 10 clients will then use Remote FX. Windows 11 doesn't support Remote FX, but we're working on an alternative.
